metricflow-semantics/tests_metricflow_semantics/time/test_time_adjuster.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import annotations
+
+import datetime
+import logging
+from typing import Dict, List, Sequence, Tuple
+
+import pytest
+import tabulate
+from _pytest.fixtures import FixtureRequest
+from dbt_semantic_interfaces.test_utils import as_datetime
+from dbt_semantic_interfaces.type_enums import TimeGranularity
+from metricflow_semantics.filters.time_constraint import TimeRangeConstraint
+from metricflow_semantics.test_helpers.config_helpers import MetricFlowTestConfiguration
+from metricflow_semantics.test_helpers.snapshot_helpers import assert_str_snapshot_equal
+from metricflow_semantics.time.dateutil_adjuster import DateutilTimePeriodAdjuster
+from metricflow_semantics.time.pandas_adjuster import PandasTimePeriodAdjuster
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="session")
+def date_times_to_check() -> Sequence[datetime.datetime]:  # noqa: D103
+    date_times = []
+    # Cover regular and leap years.
+    start_date_time = datetime.datetime(year=2020, month=1, day=1)
```

Comment:
Should we check the year 2000, which looks like a leap year and is? And also 1900, which looks like a leap year, but isn't?

I hate date/time operations.....

Reply:
Both 1900 and 2000 work, but 1900 exceeds the limit of supported times, so that was left as a comment.